.pre-commit-config.yaml

```diff
@@ -0,0 +1,52 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: check-yaml
+        args: [--allow-multiple-documents]
+      - id: check-added-large-files
+      - id: check-merge-conflict
+      - id: check-case-conflict
+      - id: check-vcs-permalinks
+      - id: check-json
+      - id: pretty-format-json
+        args: [--autofix, --no-sort-keys, --no-ensure-ascii]
+      - id: mixed-line-ending
+        args: [--fix=lf]
+      - id: no-commit-to-branch
+      - id: detect-private-key
+        exclude: (^examples/|^docs/|.*_test.go$)
+
+  - repo: https://github.com/gitleaks/gitleaks
+    rev: v8.17.0
+    hooks:
+      - id: gitleaks
+
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v3.0.0
+    hooks:
+      - id: prettier
+        types:
+          - javascript
+
+  - repo: local
+    hooks:
+      - id: golang-diff
+        name: create-go-diff
+        entry: bash -c 'git diff -p origin/main > /tmp/diff.patch'
+        language: system
+        types: [go]
+        pass_filenames: false
+
+  - repo: https://github.com/golangci/golangci-lint
+    rev: v1.53.3
```

Comment:
going forward, how are we going to keep the versions of golang ci updated? and prettier too? in this file

we also set the version for golang ci here -- https://github.com/nginxinc/nginx-kubernetes-gateway/blob/3fba53dc6d75f50281ffc0a5ec6f5c673e7e6c51/.github/workflows/lint.yml#L36C15-L36C85 and here https://github.com/nginxinc/nginx-kubernetes-gateway/blob/3fba53dc6d75f50281ffc0a5ec6f5c673e7e6c51/Makefile#L82 
and for prettier, it is set in our sources  https://github.com/nginxinc/nginx-kubernetes-gateway/blob/3fba53dc6d75f50281ffc0a5ec6f5c673e7e6c51/.github/workflows/lint.yml#L51

what about versions for other repos?

Reply:
the pre-commit CI will take care of updating this file